ark/utils/io_utils.py

```diff
@@ -72,23 +72,61 @@ def list_files(dir_name, substrs=None):
     return matches
 
 
-def extract_delimited_names(files, delimiter='_', delimiter_optional=True):
-    """Create a matched-index list of fov names from a list of files/folders
+def remove_file_extensions(files):
+    """Removes file extensions and drops preceding path from a list of files
 
-    Extracts a delimited prefix for every file in a given list of files
+    Args:
+        files (list):
+            List of files to remove file extensions from.
+            Any element that doesn't have an extension is left unchanged
+
+    Raises:
+        UserWarning:
+            Some of the processed file names still contain a period
+
+    Returns:
+        list:
+            List of files without file extensions
+    """
+
+    # make sure we don't try to split on a non-existent list
+    if files is None:
+        return
 
-    Examples:
-         - 'fov2_restofthefilename.tiff' becomes 'fov2'
-         - 'fov1.tiff' becomes 'fov1'
+    # only get the file name and not the directory path leading up to it
+    names = [os.path.split(name)[1] for name in files]
+
+    # remove the file extension
+    names = [os.path.splitext(name)[0] for name in names]
+
+    # identify names with '.' in them: these may not be processed correctly
+    bad_names = [name for name in names if '.' in name]
+    if len(bad_names) > 0:
+        print(f"These files still have \".\" in them after file extension removal: "
+              f"{','.join(bad_names)}, "
+              f"please double check that these are the correct names")
+        warnings.warn("remaining periods in file names")
```

Comment:
Why have both a print statement and a warning? Does one not show up in certain locations?

Reply:
`warnings.warn` doesn't actually print the desired warning message, only provides something that you can catch and do something with. I didn't know if we were planning to write code to explicitly catch these in the future, so I left them in.

You're right, though: we need to stick to one or the other. I should open an issue that addresses these inconsistencies. My opinion is that we should drop `warnings.warn` and simply do `print` statements. Normally, I would go the other route, but these functions that throw warnings are mostly used directly in the notebooks, and we don't want to handle the warnings in the notebooks themselves.